client_test.go

```diff
@@ -0,0 +1,47 @@
+package redis
+
+import (
+	"errors"
+	"github.com/ONSdigital/dp-redis/mock"
+	"github.com/ONSdigital/dp-sessions-api/session"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestClient_Set(t *testing.T) {
+	Convey("Given redis client is set up", t, func() {
+
+		mockClient := mock.ClientManagerMock{SetFunc: func(s *session.Session) error {
+			return nil
+		}}
+
+		Convey("Where the session is valid", func() {
+			s := session.NewSession()
+			err := mockClient.Set(s)
```

Comment:
You are testing a mock here not an actual implementation. 

I think what you want to do is create new Client (your struct) implementation passing it a mocked `goredis.Client`. Then the test will be checking if `c.client.Set` is called correctly.

Your client struct currently takes `*goredis.Client` you might be better served by creating your own interface that matches this and using this in the struct instead - this gives you the freedom to pass in anything that fulfils the interface - i.e. a mock